optimum/habana/transformers/models/mpt/modeling_mpt.py

```diff
@@ -67,9 +67,12 @@ def gaudi_mpt_attention_forward(
             else:
                 key_states = torch.cat([past_key_value[0], key_states], dim=2)
                 value_states = torch.cat([past_key_value[1], value_states], dim=2)
-        past_key_value = (key_states, value_states)
+                past_key_value = (key_states, value_states)
```

Comment:
shouldn't it be actually:
```
else:
    key_states = torch.cat([past_key_value[0], key_states], dim=2)
    value_states = torch.cat([past_key_value[1], value_states], dim=2)
past_key_value = (key_states, value_states)
```

Reply:
> shouldn't it be actually:
> 
> ```
> else:
>     key_states = torch.cat([past_key_value[0], key_states], dim=2)
>     value_states = torch.cat([past_key_value[1], value_states], dim=2)
> past_key_value = (key_states, value_states)
> ```

No this is wrong! It was like this before! please look at the code in the if condition we are updating the past_key_value and there is no need to destroy and create it again

```py
            if token_idx is not None:
                past_key_value[0].index_copy_(2, token_idx - 1, key_states)
                past_key_value[1].index_copy_(2, token_idx - 1, value_states)
                key_states = past_key_value[0]
                value_states = past_key_value[1]
            else:
                key_states = torch.cat([past_key_value[0], key_states], dim=2)
                value_states = torch.cat([past_key_value[1], value_states], dim=2)
        past_key_value = (key_states, value_states)
```

Reply:
ok, thanks for clarification